src/app/public/modules/summary-action-bar/summary-action-bar-adapter.service.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  Injectable,
+  Renderer2,
+  RendererFactory2,
+  OnDestroy
+} from '@angular/core';
+
+import {
+  Observable
+} from 'rxjs/Observable';
+
+import {
+  Subscription
+} from 'rxjs/Subscription';
+
+import 'rxjs/add/observable/fromEvent';
+
+import {
+  SkyWindowRefService
+} from '@skyux/core';
+
+import {
+  SkySummaryActionBarType
+} from './types';
+
+@Injectable()
+export class SkySummaryActionBarAdapterService implements OnDestroy {
+
+  private renderer: Renderer2;
+
+  private resizeSubscription: Subscription;
+
+  constructor(
+    private rendererFactory: RendererFactory2,
+    private windowRef: SkyWindowRefService
+  ) {
+    this.renderer = this.rendererFactory.createRenderer(undefined, undefined);
+  }
+
+  public ngOnDestroy() {
+    if (this.resizeSubscription) {
+      this.resizeSubscription.unsubscribe();
+    }
+  }
+
+  public adjustWindowMarginForActionBar(destroying?: boolean): void {
```

Comment:
The `destroying` argument is slightly confusing at first glance. I think it would be better to spit this function into two, one that prepares the body element for the action bar and one that reverts any adjustments made. Something to the effect of...

```
public styleBodyElementForActionBar()

public revertBodyElementStyles()
```

Reply:
Done